codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,14 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a GO_AWAY frame was sent for the connection.
+         */
+        void goAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a GO_AWAY frame was received for the connection.
          */
-        void goingAway();
+        void goAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
I agree that the `goAway` method should be split into `sent`/`received` where present (this may need more investigation as @buchgr indicated...but we can handle this in https://github.com/netty/netty/pull/3516).  I just want to be careful about duplicating interfaces.  Having the `sent` portion makes sense because we "automatically" send GOAWAY in some situations, so the user is not always directly initiating this event.  I'm still a bit torn on duplicating the `receive` interface between `FrameListener` and `Connection.Listener`... I see the benefit of having the `Connection.Listener` being "complete" for `GOAWAY` events, but on the other hand I'm still a bit hung up on the user will be getting the event in the GOAWAY frame. This duplication leads to confusion on our end (https://github.com/netty/netty/pull/3521#discussion_r26857369) and maybe also when someone asks "which should I use", "when are they called", "what is the difference".  If we are going to keep this interface we should think about addressing the duplication in comments or documentation.